include/xgboost/parameter.h

```diff
@@ -80,6 +80,23 @@ class FieldEntry<EnumClass> : public FieldEntry<int> {  \
 }  /* namespace dmlc */
 
 namespace xgboost {
+
+template <typename P, typename Base, typename Container>
+Args UpdateAllowUnknownImpl(P* parameter, Base* base,
+                            Container const& kwargs, bool* out_changed = nullptr) {
+  static_assert(std::is_base_of<Base, P>::value, "");
+  if (parameter->GetInitialised()) {
```

Comment:
Nit: This is British spelling. An American would write `GetInitialized`. Shall we add the use of British spelling as part of the contributor's guide? My opinion: Since British English is what the rest of the world uses, we should enforce British spelling. Maybe a linter can help with this.

Some different spellings:

British | American
------|------
Initialise | Initialize
Colour | Color
Labour | Labor
Centre | Center
Metre | Meter

Reply:
@hcho3 I think it's fine to use any of them.  I don't really notice where are contributors coming from.

Reply:
I suppose a good IDE should catch a misspelled variable before it causes compilation error.

Reply:
I think it's a bit of a non-issue, it will affect workflow barely if at all.